CONTRIBUTING.md

```diff
@@ -1,41 +1,145 @@
+smaller blocks please do so.
 
-#### **Do you want to talk to us?**
+Why? Limiting the scope of commits/pull requests makes reviewing much easier.
+Because it will usually mean each commit can be evaluated independently and a
+smaller amount of commits per pull request usually also means a smaller amount
+of code to be reviewed.
 
-* We hang out in our [gitter chatroom](https://gitter.im/openSUSE/software-o-o). This is an **informal** chat, please
-  keep discussions about bugs and features on GitHub.
+### Proper Commit Messages
 
-Thanks! :green_heart: :green_heart: :green_heart:
+We are keen on proper commit messages because they will help us to maintain
+this code in the future. We define proper commit messages like this:
+
+* The title of your commit message summarizes **what** has been done
+* The body of your commit message explains **why** you have done this
+
+If the title is to small to explain **what** you have done, then you can of course
+elaborate about it in the body. Please avoid explaining *how* you have done this,
+we are developers too and we see the diff, if we do not understand something we will
+ask you in the review.
+
+Additional to **what** and **why** you should explain potential **side-effects** of
+this change, if you are aware of any.
+
+## Development Environment
+
+To isolate your host system from OSEM development we have prepared a container
+based development environment, based on [docker](https://www.docker.com/) and
+[docker-compose](https://docs.docker.com/compose/). Here's a step by step guide
+how to set it up.
+
+**WARNING**: Since we mount the repository into our container, your user id and
+the id of the user inside the container need to be the same. If your user
+id (`id -u`) is something else than `1000` you need to configure your user id
+in the variable *CONTAINER_USERID*.
+
+1. Configure the development environment (only once).
+   ```bash
+   cp docker-compose.override.yml.example docker-compose.override.yml
+   edit docker-compose.override.yml
+   ```
+1. Build the development environment (only once)
+   ```bash
+   docker-compose build --no-cache --pull
+   ```
+1. Start the development environment:
+   ```bash
+   docker-compose up --build
+   ```
+
+1. Check out the application. You can access is at http://localhost:3000. Whatever you change will have effect in the container environment. 
+
+1. Changed something? Run the tests to verify your changes!
+   ```bash
+   docker-compose run --rm software bundle exec rake test:all
+   ```
+
+1. Issue any standard `rails`/`rake`/`bundler` command
+   ```bash
+   docker-compose run --rm software bundle exec rake rubocop:auto_correct
+   ```
+
+1. Or explore the development environment:
+   ```bash
+   docker-compose exec software /bin/bash -l
+   ```
+
+## How to contribute translations
+
+Please refer to our [translation guide](https://github.com/openSUSE/software-o-o/blob/master/README.i18n).
```

Comment:
Looking at the i18n svn repo, it's really outdated, so I assume we aren't actually using it in any capacity anymore, this should be safe to remove/replace with info about weblate.